lib/oeqa/selftest/cases/testutils.py

```diff
@@ -124,12 +124,18 @@ def verifyProvisioned(testInst, machine):
     ran_ok = False
     for delay in [5, 5, 5, 5, 10, 10, 10, 10]:
         stdout, stderr, retcode = testInst.qemu_command('aktualizr-info')
-        if retcode == 0 and stderr == b'' and stdout.decode().find('Fetched metadata: yes') >= 0:
+        if retcode == 0 and stderr == b'':
             ran_ok = True
             break
         sleep(delay)
     testInst.assertTrue(ran_ok, 'aktualizr-info failed: ' + stderr.decode() + stdout.decode())
 
+    ran_ok = False
+    if stdout.decode().find('Fetched metadata: yes') < 0:
+        stdout, stderr, retcode = testInst.qemu_command('aktualizr-info --wait-until-provisioned')
+        if retcode == 0 and stderr == b'' and stdout.decode().find('Fetched metadata: yes') >= 0:
+            ran_ok = True
+            testInst.assertTrue(ran_ok, 'aktualizr-info failed: ' + stderr.decode() + stdout.decode())
```

Comment:
`ran_ok` is now unnecessary in this section. We can instead just directly test `retcode`, `stderr`, and `stdout` for the expected values with `testInst.assertIn()` like is done in the following lines.

Also, it might be a good idea to have a comment to explain what's going on here, just something like, "First loop while waiting for the device to boot. Then wait for aktualizr to provision."